catalog/imaging/imaging-transformer-chipping/src/main/java/org/codice/alliance/imaging/chip/transformer/CatalogOutputAdapter.java

```diff
@@ -56,25 +109,420 @@ public BufferedImage getImage(ResourceResponse resourceResponse) throws IOExcept
+                        .getCoordinate0MaxCol().getLatitude()})); // upper right
+        fullImageCoordinates.add(new BasicVector(new double[] {
+                originalImageSegment.getImageCoordinates()
+                        .getCoordinateMaxRowMaxCol().getLongitude(),
+                originalImageSegment.getImageCoordinates()
+                        .getCoordinateMaxRowMaxCol().getLatitude()})); // lower right
+        fullImageCoordinates.add(new BasicVector(new double[] {
+                originalImageSegment.getImageCoordinates()
+                        .getCoordinateMaxRow0().getLongitude(),
+                originalImageSegment.getImageCoordinates()
+                        .getCoordinateMaxRow0().getLatitude()})); // lower left
+
+        logVectors("full image coordinates", fullImageCoordinates);
+
+        return fullImageCoordinates;
+    }
+
+    private void setImageData(BufferedImage chip, ImageSegment chipImageSegment)
+            throws IOException {
+        byte[] jpegData = createJpg(chip);
+
+        chipImageSegment.setData(new ImageInputStreamImpl() {
+
+            private InputStream inputStream = new ByteArrayInputStream(jpegData);
+
+            @Override
+            public int read() throws IOException {
+                return inputStream.read();
+            }
+
+            @Override
+            public int read(byte[] b, int off, int len) throws IOException {
+                return inputStream.read(b, off, len);
+            }
+        });
+        chipImageSegment.setDataLength(jpegData.length);
+    }
+
+    private List<Vector> getChipCornerPixels(int sourceX, int sourceY, int selectWidth,
+            int selectHeight) {
+        List<Vector> chipCornerPixels = new LinkedList<>();
+        chipCornerPixels.add(new BasicVector(new double[] {sourceX, sourceY}));
+        chipCornerPixels.add(new BasicVector(new double[] {sourceX + selectWidth, sourceY}));
+        chipCornerPixels.add(new BasicVector(new double[] {sourceX + selectWidth,
+                sourceY + selectHeight}));
+        chipCornerPixels.add(new BasicVector(new double[] {sourceX, sourceY + selectHeight}));
+        return chipCornerPixels;
+    }
+
+    private BinaryContent nitfToBinaryContent(NitfHeader header, ImageSegment imageSegment)
+            throws IOException {
+        byte[] data;
+        File tmpFile = File.createTempFile("nitfchip-", ".ntf");
+        try {
+            new NitfCreationFlow().fileHeader(() -> header)
+                    .imageSegment(() -> imageSegment)
+                    .write(tmpFile.getAbsolutePath());
+
+            try (FileInputStream fis = new FileInputStream(tmpFile)) {
+                data = IOUtils.toByteArray(fis);
+            }
+
+        } finally {
+            if (!tmpFile.delete()) {
+                LOGGER.debug("unable to delete the temporary file '{}'", tmpFile);
+            }
+        }
+
+        return new BinaryContentImpl(new ByteArrayInputStream(data));
```

Comment:
You need to use the BinaryContentImpl constructor that also takes the mimeType
`new BinaryContentImpl(new ByteArrayInputStream(data), new MimeType("image/nitf"));`

This should cause the file extension of the chip to be set correctly. Currently, the chip is returned without an extension.